### PR TITLE
IPython notebook justifying gamma approximation in allele fraction model

### PR DESCRIPTION
**Pull request by davidbenjamin**
This closes half of issue #538.  @samuelklee can you take a look?

### CODE REVIEW

File: docs/CNVs/justifying_gamma_approximation.ipynb [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+   "source": [
+    "# alt minor likelihood of a alt counts and r ref counts \n",
+    "# given bias ~ Gamma(alpha, beta) and minor allele fraction f\n",
+    "def exact_likelihood(bias, alpha, beta, f, a, r):\n",
+    "    gamma_dist = gamma(alpha, loc = 0, scale = 1/beta)\n",
+    "    prefactor = gamma_dist.pdf(bias)\n",
+    "    numerator = (f**a) * ((1-f)**r) * (bias**r)\n",
+    "    denominator = (f + (1-f)*bias)**(a+r)\n",
+    "    return prefactor * numerator / denominator\n",
+    "    "
```
Comment (samuelklee) [inline docs/CNVs/justifying_gamma_approximation.ipynb]: whitespace

File: docs/CNVs/justifying_gamma_approximation.ipynb [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def approximate_likelihood(bias, alpha, beta, f, a, r):\n",
+    "    n = a + r\n",
+    "    w = (1-f)*(a - alpha + 1) + beta*f\n",
+    "    lambda0 = (sqrt(w * w + 4 * beta * f * (1 - f) * (r + alpha - 1)) - w) / (2 * beta * (1 - f))\n",
+    "    \n",
+    "    y = (1 - f)/(f + (1 - f) * lambda0);\n",
```
Comment (samuelklee) [inline docs/CNVs/justifying_gamma_approximation.ipynb]: unnecessary semicolons here and in 2 lines below

File: docs/CNVs/justifying_gamma_approximation.ipynb [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+   },
+   "outputs": [],
+   "source": [
+    "#the analytic integral over bias of the approximate likelihood\n",
+    "def approximate_likelihood_marginal(alpha, beta, f, a, r):\n",
+    "    n = a + r\n",
+    "    w = (1-f)*(a - alpha + 1) + beta*f\n",
+    "    lambda0 = (sqrt(w * w + 4 * beta * f * (1 - f) * (r + alpha - 1)) - w) / (2 * beta * (1 - f))\n",
+    "    \n",
+    "    y = (1 - f)/(f + (1 - f) * lambda0);\n",
```
Comment (samuelklee) [inline docs/CNVs/justifying_gamma_approximation.ipynb]: unnecessary semicolons here and in 2 lines below

**Comment by samuelklee** [comment]
looks good to me!  maybe add a few more plots with "edge cases" (i.e., f -> 0, a = 0, r = 0) to demonstrate that nothing funky happens here.

also, i like to do a fresh reevaluation of all cells before committing an ipython notebook, just so the cell numbers are in order (but this is just ocd on my part...)

if #538 automatically closes, be sure to open a new issue for @mbabadi's half.

File: docs/CNVs/justifying_gamma_approximation.ipynb (hunk 1 above)
Comment (davidbenjamin) [inline docs/CNVs/justifying_gamma_approximation.ipynb]: done

File: docs/CNVs/justifying_gamma_approximation.ipynb (hunk 2 above)
Comment (davidbenjamin) [inline docs/CNVs/justifying_gamma_approximation.ipynb]: done

File: docs/CNVs/justifying_gamma_approximation.ipynb (hunk 3 above)
Comment (davidbenjamin) [inline docs/CNVs/justifying_gamma_approximation.ipynb]: done

**Pull request merged**